pkg/controller/capacity/capacity_controller.go

```diff
@@ -216,51 +212,60 @@ func (c *Controller) capacityTargetSyncHandler(key string) error {
 		targetDeployment, err := c.findTargetDeploymentForClusterSpec(clusterSpec, targetNamespace, selector, &clusterStatus)
 		if err != nil {
 			clusterErrors.Append(err)
-			continue
 		}
 
-		// Get the requested percentage of replicas from the capacity object. This is
-		// only set by the scheduler.
-		replicaCount := int32(replicas.CalculateDesiredReplicaCount(uint(clusterSpec.TotalReplicaCount), float64(clusterSpec.Percent)))
+		var availableReplicas, achievedPercent int32
 
-		// Patch the deployment if it doesn't match the cluster spec.
-		if targetDeployment.Spec.Replicas == nil || replicaCount != *targetDeployment.Spec.Replicas {
-			_, err = c.patchDeploymentWithReplicaCount(targetDeployment, clusterSpec.Name, replicaCount, &clusterStatus)
+		if targetDeployment != nil {
+			// Get the requested percentage of replicas from the capacity object. This is
+			// only set by the scheduler.
+			replicaCount := int32(replicas.CalculateDesiredReplicaCount(uint(clusterSpec.TotalReplicaCount), float64(clusterSpec.Percent)))
```

Comment:
Type casting looks a bit clumsy in this context. Do you think it's worth updating the method signature?

Reply:
I got started on this, but the diff was getting really long really fast... I might leave that for another day :'(